File: sdk/js/examples/transaction_request.mjs
#!/usr/bin/env node
import { SentinelVoiceClient } from '../src/index.js';

const baseUrl = process.env.SV_BASE_URL || 'http://127.0.0.1:8081';
const apiKey = process.env.SV_API_KEY || 'sv_live_REPLACE_ME';

const client = new SentinelVoiceClient({ baseUrl, apiKey });
const result = await client.createTransactionRequest({
  actionType: 'WIRE_TRANSFER',
  amountInr: 250000,
  beneficiaryRef: 'BEN-001',
  requesterEmployeeRef: 'E1001',
  channel: 'CORE_BANKING',
});
console.log(JSON.stringify(result, null, 2));

File: sdk/python/examples/transaction_request.py
#!/usr/bin/env python3
"""Runnable example: pre-transaction gate check."""
import os
import sys

# Allow running without install
sys.path.insert(0, os.path.join(os.path.dirname(__file__), ".."))

from sentinelvoice import SentinelVoiceClient


def main() -> None:
    base = os.environ.get("SV_BASE_URL", "http://127.0.0.1:8081")
    key = os.environ.get("SV_API_KEY", "sv_live_REPLACE_ME")
    client = SentinelVoiceClient(base_url=base, api_key=key)
    result = client.create_transaction_request(
        action_type="WIRE_TRANSFER",
        amount_inr=250_000,
        beneficiary_ref="BEN-001",
        requester_employee_ref="E1001",
        channel="CORE_BANKING",
    )
    print(json_dumps(result))


def json_dumps(obj):
    import json

    return json.dumps(obj, indent=2)


if __name__ == "__main__":
    main()
